app/src/lib/stores/app-store.ts

```diff
@@ -4821,7 +4840,20 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
     // skip pushing if the current branch is a detached HEAD or the repository
     // is unborn
-    if (gitStore.tip.kind === TipState.Valid) {
+    if (
+      gitStore.tip.kind === TipState.Valid &&
+      gitStore.defaultBranch !== null &&
+      gitStore.tip.branch.name !== gitStore.defaultBranch.name
+    ) {
```

Comment:
I think these two lines should be removed to preserve the previous behavior (and they're already in the nested `if` anyway, which I guess is where they belong)

```suggestion
    if (gitStore.tip.kind === TipState.Valid) {
```